librosa/core/constantq.py

```diff
@@ -172,46 +176,32 @@ def cqt(y, sr=22050, hop_length=512, fmin=None, n_bins=84,
 
     # Determine required resampling quality
     Q = float(filter_scale) / (2.0**(1. / bins_per_octave) - 1)
-
     filter_cutoff = fmax_t * (1 + filters.window_bandwidth('hann') / Q)
-
     nyquist = sr / 2.0
-
     if filter_cutoff < audio.BW_FASTEST * nyquist:
         res_type = 'kaiser_fast'
-    elif filter_cutoff < audio.BW_BEST * nyquist:
-        res_type = 'kaiser_best'
     else:
         res_type = 'kaiser_best'
 
-    cqt_resp = []
-
-    len_orig = len(y)
-
     y, sr, hop_length = __early_downsample(y, sr, hop_length,
-                                           res_type, n_octaves,
+                                           res_type,
+                                           n_octaves,
                                            nyquist, filter_cutoff)
 
-    n_filters = min(bins_per_octave, n_bins)
+    cqt_resp = []
 
     if res_type != 'kaiser_fast':
 
-        # Do two octaves before resampling to allow for usage of sinc_fastest
-        fft_basis, n_fft, filter_lengths = __fft_filters(sr, fmin_t,
-                                                         n_filters,
-                                                         bins_per_octave,
-                                                         tuning,
-                                                         filter_scale,
-                                                         norm,
-                                                         sparsity)
-        min_filter_length = np.min(filter_lengths)
+        # Do the top octave before resampling to allow for fast resampling
+        fft_basis, n_fft, _ = __fft_filters(sr, fmin_t,
+                                            n_filters,
+                                            bins_per_octave,
+                                            tuning,
+                                            filter_scale,
+                                            norm,
+                                            sparsity)
 
-        # Compute a dynamic hop based on n_fft
-        my_cqt = __variable_hop_response(y, n_fft,
-                                         hop_length,
-                                         min_filter_length,
-                                         fft_basis,
-                                         aggregate)
+        my_cqt = __fft_response(y, n_fft, hop_length, fft_basis)
 
         # Convolve
```

Comment:
You say # Convolve in two places, but both times the convolution actually happened on the preceding line, and doesn't need a comment anyway.